src/actix/mod.rs

```diff
@@ -142,6 +146,7 @@ pub fn init(
                 .app_data(validate_json_config)
                 .app_data(TempFileConfig::default().directory(&upload_dir))
                 .app_data(MultipartFormConfig::default().total_limit(usize::MAX))
+                .app_data(hardware_reporting_settings.clone())
```

Comment:
Extra argument just for this bool seems like an overkill.

How about we either put it into global static, or pass the whole settings object?